overwatch-rs/src/services/state.rs

```diff
@@ -20,16 +20,26 @@ pub trait ServiceState: Sized {
     type Settings;
     /// Errors that can occur during state initialization
     type Error;
-    /// Initialize a stage upon the provided settings
+    /// Initialize a state using the provided settings.
+    /// This is called when [StateOperator::try_load] doesn't return a state.
     fn from_settings(settings: &Self::Settings) -> Result<Self, Self::Error>;
 }
 
 /// A state operator is an entity that can handle a state in a point of time
 /// to perform any operation based on it.
+/// A typical use case is to handle recovery: Saving and loading state.
 #[async_trait]
 pub trait StateOperator {
     /// The type of state that the operator can handle
     type StateInput: ServiceState;
+    /// Errors that can occur during state loading
+    type LoadError;
```

Comment:
Should we bind this to `Error`? I think its probably a smart choice. Otherwise we may end up not able to use `?` in some cases.

Reply:
Probably both of them. But let the change for a further PR in the `ServiceState`, as it may introduce more breaking changes. Open an issue though. Thanks!